cgo/types.go

```diff
@@ -0,0 +1,530 @@
+package cgo
+
+/*
+#cgo LDFLAGS: -L${SRCDIR}/..
+#cgo pkg-config: ${SRCDIR}/../filcrypto.pc
+#include "../filcrypto.h"
+#include <stdlib.h>
+*/
+import "C"
+import (
+	"unsafe"
+)
+
+type FCPResponseStatus int64
+
+type SliceBoxedUint8 = C.struct_slice_boxed_uint8
+
+func (ptr SliceBoxedUint8) Slice() []byte {
+	return unsafe.Slice((*byte)(ptr.ptr), int(ptr.len))
+}
+
+func (ptr SliceBoxedUint8) Copy() []byte {
+	if ptr.len == 0 {
+		return []byte{}
+	}
+
+	res := make([]byte, int(ptr.len))
+	copy(res, ptr.Slice())
+	return res
+}
+
+func (ptr SliceBoxedUint8) Destroy() {
+	C.destroy_boxed_slice(ptr)
+}
+
+type SliceRefUint8 = C.slice_ref_uint8_t
+
+type BLSDigest = C.BLSDigest_t
+
+func (ptr *BLSDigest) Slice() []byte {
+	return ptr.inner.Slice()
+}
+
+type HashResponse = C.HashResponse_t
+
+func (ptr *HashResponse) Digest() []byte {
+	return ptr.digest.Slice()
+}
```

Comment:
For stuff like this, I'd consider either:

1. Copying.
2. Returning the "desired" type (in this case, `Digest`).

Reply:
cleaned up